// ng2-growl/src/growl.component.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var growl_service_1 = require("./growl.service");
var GrowlComponent = (function () {
    function GrowlComponent(growlService) {
        var _this = this;
        this.growlService = growlService;
        this.types = growl_service_1.Types;
        this.growlItems = [];
        this.growlService.OnAddMessage.subscribe(function (msg) { return _this.onAddMessage(msg); });
        this.growlService.OnRemoveMessage.subscribe(function (msg) { return _this.onRemoveMessage(msg); });
    }
    GrowlComponent.prototype.ngOnInit = function () {
    };
    GrowlComponent.prototype._remove = function (msg) {
        var _this = this;
        if (!msg || !this.growlItems || !this.growlItems.length)
            return;
        var curMsg;
        var curMsgIdx;
        for (curMsgIdx = 0; curMsgIdx < this.growlItems.length; curMsgIdx++) {
            if (this.growlItems[curMsgIdx] && this.growlItems[curMsgIdx].id === msg.id) {
                curMsg = this.growlItems[curMsgIdx];
                break;
            }
        }
        if (!curMsg || curMsg.closing)
            return;
        curMsg.closing = true;
        setTimeout(function () {
            _this.growlItems.splice(curMsgIdx, 1);
        }, 500);
    };
    GrowlComponent.prototype.close = function (index) {
        if (typeof index !== 'number')
            return;
        this._remove(this.growlItems[index]);
    };
    GrowlComponent.prototype.onAddMessage = function (msg) {
        this.growlItems.push(msg);
    };
    GrowlComponent.prototype.onRemoveMessage = function (msg) {
        this._remove(msg);
    };
    return GrowlComponent;
}());
GrowlComponent = __decorate([
    core_1.Component({
        selector: 'ng2-growl',
        template: "\n<div class=\"message-container {{types[item.type].toLowerCase()}}\" *ngFor=\"let item of growlItems; let idx = index;\" [ngClass]=\"{closing: item.closing}\">\n  <span class=\"close\" (click)=\"close(idx)\">\u00D7</span>\n  <strong class=\"heading\">{{item.heading}}</strong>\n  <span class=\"message\">{{item.message}}</span>\n</div>\n",
        styles: ['@keyframes slideIn{0%{transform:translateX(100%)}100%{transform:translateX(0)}}',
            '@keyframes slideOut{0%{transform:translateX(0)}100%{transform:translateX(100%)}}',
            ':host{position:fixed;top:0;right:0;z-index:500}',
            '.message-container{transform-origin:top right;animation:.3s ease-out 0s 1 forwards slideIn;position:relative;padding:1rem;margin:.75rem 0;width:27rem;background-color:rgba(250,250,250,.9);border:.15rem solid #e8e8e8;border-radius:.5rem 0 0 .5rem;box-shadow:-.1rem .1rem .5rem #666}',
            '.message-container.closing{animation:.3s ease-out 0s 1 forwards slideOut}',
            '.message-container.error{border-left:.5rem solid #BC5543}',
            '.message-container.warn{border-left:.5rem solid #E7E232}',
            '.message-container.info{border-left:.5rem solid #146587}',
            '.message-container.success{border-left:.5rem solid #477529}',
            '.close{position:absolute;top:.5rem;right:.5rem;cursor:pointer;font-size:2rem;font-weight:700;line-height:0;padding:.5rem .2rem}',
            '.heading{display:block}']
    }),
    __metadata("design:paramtypes", [growl_service_1.GrowlService])
], GrowlComponent);
exports.GrowlComponent = GrowlComponent;
